### DISCUSSION TITLE
Getting the PortAlignment where link is attached to ?

### DISCUSSION THREAD

**Discussion by Jukser** [Blazor-Diagrams/Blazor.Diagrams#440] [Q&A]
Hi

Been very happy with Blazor-Diagram, it suites my needs just perfectly.

But now I faced a problem that I cannot figure out.
How to get to which side of node the link is attached to?  ie. PortAlignment.Right or PortAlignment.Left, etc.
Need this info in other parts of my app. (from both ends, source and target)

I was trying to get the sources PortAligment in BusinessDiagram.Links.Added event  but just couldn't figure out how to access this info trough BaseLinkModel.

It really doesn't have to be in the Event, just need to access this info at some point.
Any help is appreciated. :)

Cheers
Juha

**Comment by PiggyChu620** [comment]
If I understand your question right, you could get it from

PortAlignment alignment;
PortModel port = theLink.Source.Model as PortModel;    //Or theLink.Target.Model as PortModel
if(port !=null) alignment = port.Alignment;
